lib/Utility.js

```diff
@@ -33,10 +33,10 @@ var Utility = module.exports = {
   },
   extendExtraConfig : function(proc, opts) {
     if (opts.env && opts.env.current_conf) {
-      if (opts.env.current_conf.env &&
-          typeof(opts.env.current_conf.env) === 'object' &&
-          Object.keys(opts.env.current_conf.env).length === 0)
-        delete opts.env.current_conf.env
+      // NOTE: `env` gets overriden by `Utility.extendMix` call. Instead, we want to merge envs.
+      if (opts.env.current_conf.env && typeof(opts.env.current_conf.env) === 'object')
+        Utility.extendMix(proc.pm2_env.env, opts.env.current_conf.env);
```

Comment:
AFAIK no, because the env won't be extended with the new variables here:

https://github.com/Unitech/pm2/blob/304fc5e59706bd26dbd8c6660e6e1b505ceff487/lib/API.js#L1362